### PR TITLE
Pin openssl to 1.1.1n

### PR DESCRIPTION
**Pull request by LarryOsterman**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

**Review by gearama — APPROVED** [review]

**Review by antkmsft — APPROVED** [review]

File: vcpkg.json [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
+    {
+      "name": "openssl"
+    }
+  ],
+  "overrides": [
+    {
+      "name": "openssl",
+      "version-string": "1.1.1n"
+    }
+  ]
```
Comment (vhvb1989) [inline vcpkg.json:22]: This file is **only** for building the SDk locally from source (using VCPKG)

I am curious about what's the story about people using VCPKG to install a service (letting VCPKG build the package)

File: vcpkg.json [hunk 2]
```diff
@@ -1,16 +1,23 @@
...
-            "name": "libxml2",
-            "platform": "!windows & !uwp"
-        },
-        {
-            "name": "openssl"
-        }
-    ]
+  "name": "azure-sdk-for-cpp",
+  "version-string": "1.0.0",
+  "builtin-baseline": "f0aa678b7471497f1adedcc99f40e1599ad22f69",
```
Comment (vhvb1989) [inline vcpkg.json:4]: I wish there were better docs for this.
I see this: https://github.com/microsoft/vcpkg/blob/master/docs/users/versioning.md#builtin-baseline
But I don't fully understand what's happening.
Is this a VCPKG repo's commit Id?  or an azure-sdk-cpp's repo commit id?  Or is this a openssl repo's commit id??

If this is a VCPKG repo's commit id, is my local vcpkg folder going to checkout that commit id before building the dependencies?

What would it happen if we have an Azure package (imagine azure-attestation-cpp) which requires azure-core-cpp with a version that is in a more recent VCPKG commit id than the one we set here for OpenSSL??   Isn't that a conflict?

Comment (LarryOsterman) [reply vcpkg.json:4]: It took me some time to figure it out. It's a vcpkg repo commit ID, and it takes the place of the vcpkg-commit.txt. Basically it establishes a baseline in vcpkg against which all packages will be pulled.

With this added, I believe we can remove support for vcpkg-commit.txt from our CI pipeline, but that's out of scope for this pull request.

Comment (LarryOsterman) [reply vcpkg.json:4]: Oh, and as best as I can figure, the way this works is that when vcpkg is executing the manifest base vcpkg install, it syncs a separate directory to that commit hash, so it doesn't affect your local vcpkg repo. However it does mean that your local vcpkg repo needs to have the commit hash in its history.

File: vcpkg.json (hunk 1 above)
Comment (LarryOsterman) [reply vcpkg.json:22]: Yes, I know - but that's the file that our CI system and we use daily when we build, so it's the one that needs to be updated.

I am having a separate discussion about what we do w.r.t. the vcpkg.json files in each package directory (which I believe are functionally ignored today) and the vcpkg.json file we ship (there are 37 different vcpkg.json files in the tree :)).

**Pull request merged**